### PR TITLE
change configuration for major version update

### PR DESCRIPTION
**Pull request by jondayton**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,92 +155,17 @@ const store = new AppStore({
...
-```
-
-### find/fetch options
-
-The `find` and `fetch` methods can take an options object as a second argument.
+### Retrieving documents
+The store uses variations of `get`, `find` and `fetch` to retrieve records from the store, with `All`, `Many`, and `One` determining how many records will be requested. An optional `options` hash provides further customization for requesting and caching.
+- `get` will only retrieve locally cached records, and will never request from the server
+- `find` will first look for records of a given type in the store, and then will go to the server
+- `fetch` will always requet from the server
```
Comment (anitacode) [inline README.md:162]: requet to request

File: spec/Model.spec.js [hunk 2]
```diff
@@ -414,231 +355,456 @@ describe('Model', () => {
...
       })
 
-      store.createOrUpdateModelFromData({
-        type: 'notes',
-        id: 11,
-        attributes: { description: 'Note 2 for Todo 100' },
-        relationships: { todo: { data: { type: 'todos', id: 100 } } }
+      it('relationship can be set', () => {
+        note.todo = todo
+        expect(note.todo).toEqual(todo)
```
Comment (anitacode) [inline spec/Model.spec.js:722]: Duplicated

File: spec/Model.spec.js [hunk 3]
```diff
@@ -414,231 +355,456 @@ describe('Model', () => {
...
-        type: 'todos',
-        id: 100,
-        attributes: { description: 'Todo 100' },
-        relationships: { notes: { included: false } }
+      it('relationship can be unset', () => {
+        note.todo = todo
+        expect(note.todo).toEqual(todo)
+
+        note.todo = null
+        expect(note.todo).toBeFalsy()
```
Comment (anitacode) [inline spec/Model.spec.js:730]: Duplicated

File: spec/Store.spec.js [hunk 4]
```diff
@@ -1593,20 +1606,31 @@ describe('Store', () => {
...
 
         fetch.mockResponseOnce(createMockTodosResponse(100, '1000'))
         fetch.mockResponseOnce(createMockTodosResponse(75, '1100'))
 
         store.add('todos', createMockTodosAttributes(150, '1175'))
 
+        // a Todo with id 1174 is added but with no attributes
+        store.add('notes', {
+          todo: {
+            id: '1174',
```
Comment (anitacode) [inline spec/Store.spec.js:1619]: Should be 1173??

File: spec/Store.spec.js [hunk 5]
```diff
@@ -1593,20 +1606,31 @@ describe('Store', () => {
...
         const todos = await store.findMany('todos', ids)
 
         expect(todos).toHaveLength(300)
         expect(store.getAll('todos')).toHaveLength(325)
 
         expect(fetch.mock.calls).toHaveLength(2)
+        expect(
+          fetch.mock.calls.some((call) => call[0].match(/1173/))
+        ).toBeTruthy()
         expect(
```
Comment (anitacode) [inline spec/Store.spec.js:1634]: Double check. :)

File: src/relationships.js [hunk 6]
```diff
@@ -124,19 +143,27 @@ export const defineToManyRelationships = action((record, store, toManyDefinition
  * @returns {object} the related record
  */
 export const setRelatedRecord = action((relationshipName, record, relatedRecord, store, inverse) => {
+  if (record == null) { return null }
```
Comment (anitacode) [inline src/relationships.js:146]: Continuing here.

**Review by anitacode — APPROVED** [review]

File: spec/Store.spec.js (hunk 4 above)
Comment (jondayton) [reply spec/Store.spec.js:1636]: this is actually correct.

File: spec/Model.spec.js (hunk 3 above)
Comment (jondayton) [reply spec/Model.spec.js:747]: I'm going to keep it. testing in different contexts.

File: README.md (hunk 1 above)
Comment (jondayton) [reply README.md]: updated.

**Pull request merged**